SimplenoteIntents/IntentHandler.swift

```diff
@@ -2,18 +2,53 @@ import Intents
 import CoreData
 
 class IntentHandler: INExtension {
+    let coreDataManager: CoreDataManager
+    let dataController: WidgetDataController
+
+    override init() {
+        do {
+            self.coreDataManager = try CoreDataManager(StorageSettings().sharedStorageURL, for: .intents)
+            self.dataController = try WidgetDataController(coreDataManager: coreDataManager)
+        } catch {
+            fatalError()
+        }
+        super.init()
+    }
+
+
     override func handler(for intent: INIntent) -> Any {
         return self
     }
 }
 
 extension IntentHandler: NoteWidgetIntentHandling {
     func provideNoteOptionsCollection(for intent: NoteWidgetIntent, with completion: @escaping (INObjectCollection<WidgetNote>?, Error?) -> Void) {
-        var notes: [WidgetNote] = []
 
-        // TODO: add intent core data implementation.
+        // Fetch notes
+        var notes: [Note] = []
+        do {
+            notes = try dataController.notes()
+        } catch {
+            completion(nil, error)
+        }
```

Comment:
Question: where is the `completion` closure used? is the Error actually passed over to the user?

- If the error is presented somehow, we should prob. have our own (rather than relaying the Core Data one).
- If not, we could just drop it, and pass an empty array?

```
guard let notes = try? dataController.notes() else {
     completion([])
     return
}
```

Reply:
What appears to the user is a generic error at widget configuration.  I don't think the user is presented any of our details, but it tells the configurator there is an error and then logs the error output.  Returning an empty array would allow the configuration to load but without their notes.

🤔 let me break the widgets so I can show this

Reply:
ah, it does display:
<img height="400" src="https://cdn-std.droplr.net/files/acc_593859/OYtqFz" />

Reply:
Will add some sort of error case to be handled for this